src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -804,7 +804,7 @@ internal RegexNode ScanGroupOpen()
 
                                 // grab part after - if any
 
-                                if ((capnum != -1 || proceed == true) && CharsRight() > 0 && RightChar() == '-')
+                                if ((capnum != -1 || proceed == true) && CharsRight() > 1 && RightChar() == '-')
```

Comment:
The fix. There are no other cases of MoveRight then RightChar not already protected.